corehq/apps/hqwebapp/tables/pagination.py

```diff
@@ -39,6 +40,76 @@ def get_paginate_by(self, table_data):
         return self.current_paginate_by
 
 
+class HtmxInvalidPageRedirectMixin:
+    """
+    This mixin is used to handle the case when the page number served by
+    the view is out of range (perhaps when the queryset's filter changes).
+
+    The default behavior of `paginate_queryset` on a `ListView` raises
+    an `Http404` exception. This mixin will catch that exception and
+    redirect to the first page of the TableView instead.
+
+    IMPORTANT: THIS MIXIN MUST BE THE FIRST IN THE CLASS HIERARCHY.
+    Specifically, before `SelectablePaginatedTableView`, which it is often
+    used with.
+
+    e.g.:
+        class MyTableView(HtmxInvalidPageRedirectMixin, SelectablePaginatedTableView):
+            ...
+    """
+
+    def paginate_queryset(self, queryset, page_size):
+        try:
+            return super().paginate_queryset(queryset, page_size)
+        except Http404:
+            # This is a workaround for the case when the queryset is empty.
+            # The default behavior of `paginate_queryset` raises an Http404
+            # exception when the page number is out of range.
+            raise InvalidPage
+
+    def get_host_url(self):
+        """
+        This is used to set the `HX-Push-Url` header in the `get_first_page` response.
+
+        :return: str
+            The url of the view that is hosting the table view.
+        """
+        raise NotImplementedError(
+            "please return a URL of the view hosting the table view in the "
+            "get_host_url() method"
+        )
+
+    def get_first_page(self, request, *args, **kwargs):
+        """
+        We re-render the response instead of a redirect (HttpResponseRedirect)
+        because the HX-Push-Url header is ignored by the browser when a redirect
+        occurs.
+        """
+        # sort is the only GET parameter we want to keep
```

Comment:
I'd have expected this to just pop off the page (or reset it to 1). Are there other pagination-related parameters?

Reply:
i wouldn't call `sort` a paginated related parameter. It is a table view parameter. There aren't others i'm aware of right now

Reply:
Got it. It'd be a little more future-proof to just remove/replace the one parameter you're concerned with rather than clear out the parameters.